eiffellib/events/eiffel_base_event.py

```diff
@@ -169,14 +169,31 @@ class EiffelBaseEvent(object):
 
     schema_file = None
     __schema = None
+    __routing_key = "eiffel.{family}.{type}.{tag}.{domain_id}"
+    family = None
+    tag = None
+    domain_id = None
     version = "0.0.1"
     meta = EiffelBaseMeta("EiffelBaseEvent", version)
     links = EiffelBaseLink()
 
-    def __init__(self, version=None):
-        """Initialize with a base data object."""
+    def __init__(self, version=None, family="_", tag="_", domain_id="_"):
+        """Initialize with a base data object.
+
+        :param version: If not None use this version when loading json schemas.
+        :type version: str
+        :param family: Routing key family as per the sepia recommendation. Defaults to "_".
+        :type family: str
+        :param tag: Routing key tag as per the sepia recommendation. Defaults to "_".
+        :type tag: str
+        :param domain_id: Routing key domain id as per the sepia recommendation. Defaults to "_".
+        :type domain_id: str
+        """
         if version is not None:
             self.version = version
+        self.family = family
+        self.tag = tag
+        self.domain_id = domain_id
         self.data = EiffelBaseData()
         self.load_schema(self.version)
```

Comment:
How is self.version populated when not given as a kwarg? I can't find any logic for picking the most recent version of the schema.

Reply:
It is populated by the event classes that inherit `EiffelBaseEvent` https://github.com/eiffel-community/eiffel-pythonlib/blob/main/eiffellib/events/eiffel_activity_canceled_event.py#L35

Reply:
Aha, gotcha.